[GRUPO N°3][Segundo parcial] Entrega final

Tomig02 opened the pull request:
# 📦 [GRUPO N°3][Parcial N°2] Entrega de la actividad


## 🧾 Descripción General de la Entrega

Esta Pull Request representa la **entrega final del parcial N°2**. Incluye la integración de los aportes realizados por todos los integrantes del grupo, previamente mergeados en `develop`.

---

## 👥 Integrantes del Grupo

| Nombre completo | Rol asignado en esta entrega | Usuario GitHub |
|-----------------|-------------------------------|----------------|
| Thallys Gomes Leandr        |  Desarrollador JS Asíncrono + Tester QA/JS   | @Thallys8      |
| Guardo Tomas Agostino        | Coordinador/DevOps + Desarrollador JS librerias externas | @Tomig02      |

---

## 📂 Archivos y entregables principales incluidos

- `readme.md` Actualizado con la informacion de la entrega
- `changelog.md` Con las contribuciones de cada integrante
- `script.js` Con las adaptaciones a la implementación de DOM
- `docs/04-diagramas/` Nuevos diagramas POO
- `js/` Con la implementacion de los flujos JS
- `js/test/` Con los tests de la implementacion en JS
-  `js/model/` Con la implementacion de los modelos DOM
- `js/utils/` Con la implemantación del Storage
- `index.html` Actualizado al agregar nuevas funciones al sistema

---

## 📋 Checklist de revisión antes de merge

- [x] Todas las ramas `feature/*` fueron mergeadas en `develop`.
- [x] Se resolvieron todos los conflictos.
- [x] La estructura de carpetas y archivos sigue el formato solicitado.
- [x] Se completó correctamente el archivo `changelog.md`.
- [x] Se revisaron los comentarios de revisión en cada PR individual.
- [x] Esta Pull Request se realiza desde `release/primer-parcial` hacia `master`.
- [x] Se notificó en el canal de Slack con el formato correcto.

---

## 📣 Comunicación en Slack

Una vez creada esta Pull Request, notificamos en el canal:

[Entrega Obligatoria N°3][GRUPO 3](enlace a esta Pull Request)


Posteriormente iniciamos un hilo mencionando a @Matias Velasquez, indicando el contenido general de la entrega.  
Usaremos los siguientes emojis durante la revisión:

- 💬 `:speech_balloon:` → Comentarios pendientes
- ✅ `:white_check_mark:` → Correcciones realizadas
- 🟢 `:lgtm:` → Aprobado para merge por el docente (LGTM)
- 🔀 `:twisted_rightwards_arrows:` → Merge completado (debe ser notificado por el grupo)

---

## 📌 Notas adicionales

- Esta PR **no debe mergearse automáticamente**.
- El merge hacia `master` **solo se realiza una vez recibida la aprobación oficial (LGTM) por parte del docente en GitHub**.
- El contenido mergeado en `master` será considerado como **la versión final en producción**

Tomig02 committed:
Integracion de libreria AOS al sistema actual

Tomig02 committed:
documentacion de nueva libreria

Tomig02 committed:
actualizacion de readme.md

Tomig02 committed:
documento en changelog.md

Tomig02 committed:
Merge PR #117 desde backport/release-cuarta-entrega - Backport completo

Backport de la release por la cuarta entrega

Tomig02 committed:
Merge PR #121 desde feature/dev-libreria-externa-AOS - Implementacion y documentacion

Integracion de libreria externa AOS

Tomig02 committed:
agrego al changelog.md

Tomig02 committed:
Merge branch 'develop' into feature/coordinador-devops-readme-documentacion

Tomig02 committed:
Merge PR #123 desde feature/coordinador-devops-readme-documentacion - Actualizacion Readme.md

Actualizacion de readme.md

Thallys8 committed:
Creando el codigo async fet-api

Thallys8 committed:
creando la prueba bise line con lighthouse

Thallys8 committed:
agregando numedo de la issue del caso de prueba

Thallys8 committed:
actualizando numero de la issue

Thallys8 committed:
agregando los casos de pueba con post fetch y post library

Thallys8 committed:
Add files via upload

Agregando las capturas de pantalla de los test case 11, 12 y 13

Thallys8 committed:
Agregando la prueba jasmine de API

Thallys8 committed:
Cgregando la librar.spec.js

Thallys8 committed:
Update changelog.md

Actualizando changelog

Thallys8 committed:
Update changelog with new features and changes

Actualizando changelog

Thallys8 committed:
Update changelog.md

Tomig02 committed:
Merge PR #125 fdesde feature/dev-async-fetch-api

Feature/dev-async-fetch-api agregando los archivos de fetch api

Tomig02 committed:
Merge branch 'develop' into feature/tester-qa-js-testing-suite

Tomig02 committed:
Merge pr #126 desde feature/tester-qa-js-testing-suite

Feature/tester-qa-js-testing-suite agregando pruebas Jamine y Lighthouse

MVelasquez98 reviewed (changes requested):
Sin el deploy en GithubPages no puedo corregir la entrega:
<img width="913" height="141" alt="image" src="https://github.com/user-attachments/assets/4473c969-55d1-411d-ae80-ddd90ed650cb" />

Tomig02 commented:
> Sin el deploy en GithubPages no puedo corregir la entrega: <img alt="image" width="913" height="141" src="https://private-user-images.githubusercontent.com/43156596/518654868-4473c969-55d1-411d-ae80-ddd90ed650cb.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQwNzgzNzgsIm5iZiI6MTc2NDA3ODA3OCwicGF0aCI6Ii80MzE1NjU5Ni81MTg2NTQ4NjgtNDQ3M2M5NjktNTVkMS00MTFkLWFlODAtZGRkOTBlZDY1MGNiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTI1VDEzNDExOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdmZTM5N2Q5Y2YyNTVmOWQzZjcxZGUxOWFmZWI4MzMzZjUzNzNlZTllNmI4ODc0ZmQ0MDk4NTY5MTdmNjFjYzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.wIU35DS2G9i09vpZG6CXeJyx9e5sK74tT0HYc09w6iM">

Solicitado a @Thallys8

MVelasquez98 commented on docs/03-testing/test-case-11-lighthouse-baseline.md:
```diff
@@ -0,0 +1,41 @@
+![Captura Accessibility](./screenshots/lighthouse-baseline-accessibility.png) 
+
+### Best Practices: [96] 
+- [Listar observaciones] 
```
¿Y cuales son?

MVelasquez98 commented on docs/03-testing/test-case-11-lighthouse-baseline.md:
```diff
@@ -0,0 +1,41 @@
+- [Listar observaciones]
+
+## Issues Generadas 
+- [#124] Optimizar tamaño de imágenes para mejorar LCP y CLS
```
Nunca se hizo nada con esta issue, no hay PR asociada con solución. Simplemente se abrió y cerro

MVelasquez98 commented on js/api/apiService.js:
```diff
@@ -0,0 +1,57 @@
+    },
+
+    showLoading() { 
+        // Actualizar UI con estado de carga 
```
Estan sin implementar estas funciones

MVelasquez98 commented on js/api/apiService.js:
```diff
@@ -0,0 +1,57 @@
+    }, 
+
+    showError(message) { 
+        // Mostrar mensaje de error en UI
```
La consola no es la UI

MVelasquez98 commented on js/test/test-runner.html:
```diff
@@ -25,13 +25,15 @@ <h2 style="text-align:center; font-family:sans-serif;">Pruebas Jasmine</h2>
     <!-- Tests --> 
     <script type="module" src="script.spec.js"></script>
     <script type="module" src="models.spec.js"></script>
+    <script type="module" src="api.spec.js"></script>
```
¿Y el de library no lo probaste?

MVelasquez98 commented on js/api/apiService.js:
```diff
@@ -0,0 +1,57 @@
+     * @returns {Promise<Object>} Datos de usuarios obtenidos 
+     */ 
+    async fetchUsers() {
+        const endpoint = 'https://jsonplaceholder.typicode.com/users';
```
Nunca usas esta api externa, no tiene sentido. La consigna solicita:
```
También puedes utilizar un archivo JSON estático incluido en el proyecto, incluyendo dentro de una carpeta “api”.
La información estática o obtenida de una API debe cargarse y utilizarse de forma asíncrona.
```
¿para que usuarios?¿por que no las actividades? El uso de la api "externa" tiene que tener sentido con el proyecto

MVelasquez98 commented on js/test/api.spec.js:
La consigna solicita:
<img width="541" height="96" alt="image" src="https://github.com/user-attachments/assets/bfa52486-ca9a-4ede-b4a3-ce0271eff009" />
¿Como se si estos test corrieron bien?¿Generaste issues si algun test no funciono?

MVelasquez98 reviewed (changes requested):
# Tabla de evaluación (global + individual)

| **Estudiante** | **Rol Asignado** | **Puntaje Rol (máx. 2,5)** |
|---|---|---|
| Thallys Gomes Leandro | Desarrollador JS Asíncrono  | [ 0,7 puntos ] |
| Guardo Tomas Agostino | Desarrollador JS librerias externas | [ 2,5 puntos ] |
| Thallys Gomes Leandro | Tester QA/JS - Testing Avanzado | [ 1 punto ] |
| Guardo Tomas Agostino | Coordinador/DevOps | [ 0,9 puntos ] |

---

# Ítems por Rol – cada uno vale hasta 2,5 puntos

---

# 1. Desarrollador JS Asíncrono - Fetch & APIs (máx. 2,5 puntos)

### Implementación de consumo asíncrono (1,3 puntos)

- **[0 pts]** ❌ Selecciona y justifica API utilizada (externa o JSON estático), documenta decisión técnica  
- **[0,4 pts]** ✅ Implementa funciones asíncronas con fetch y promises correctamente estructuradas  
- **[0 pts]** ❌ Maneja estados de carga (loading, success, error) con feedback visual en UI  
- **[0 pts]** ❌ Integra datos obtenidos con el DOM dinámicamente de forma coherente  
  > 💬 *Comentario: La api que se consume, nunca se utiliza en el proyecto. Esta aislada del proyecto.*

### Procesamiento de datos (0,5 puntos)

- **[0 pts]** ❌ Aplica funciones de orden superior (map, filter, reduce)  
- **[0 pts]** ❌ Valida y sanitiza datos recibidos correctamente  

### Manejo de errores (0,3 puntos)

- **[0,2 pts]** ✅ Implementa try-catch en operaciones asíncronas  
- **[0 pts]** ❌ Muestra mensajes de error amigables al usuario  

### Integración y arquitectura (0,2 puntos)

- **[0 pts]** ❌ Coordina con clases POO para almacenar datos  
- **[0 pts]** ❌ Mantiene separación de responsabilidades  

### Participación técnica (0,2 puntos)

- **[0,05 pts]** ✅ Crea rama feature/dev-async-fetch-api  
- **[0,05 pts]** ✅ Documenta en changelog.md  
- **[0,025 pts]** ⚠️PR + merge revisado por Coordinador  
- **[0,025 pts]** ⚠️ Responde issues del Tester  
  > 💬 *Comentario: No hay issue asociada a la PR*

---

# 2. Desarrollador JS Librerías Externas (máx. 2,5 puntos)

### Selección y justificación (0,5 puntos)

- **[0,3 pts]** ✅ Selecciona librería adecuada para una necesidad real  
- **[0,2 pts]** ✅ Justifica técnicamente la elección  

### Integración técnica (0,8 puntos)

- **[0,4 pts]** ✅ Instala/integra librería (CDN)  
- **[0,2 pts]** ✅ Configura según documentación  
- **[0,2 pts]** ✅ Crea wrapper si es necesario  

### Implementación (0,5 puntos)

- **[0,3 pts]** ✅ Integra librería en 2 partes de la app  
- **[0,2 pts]** ✅ Mantiene consistencia visual  

### Documentación (0,5 puntos)

- **[0,3 pts]** ✅ Crea /docs/06-librerias/libreria-doc.md completo  
- **[0,2 pts]** ✅ Incluye capturas de implementación  

### Participación técnica (0,2 puntos)

- **[0,05 pts]** ✅ Crea rama feature/dev-libreria-externa-[nombre]  
- **[0,05 pts]** ✅ Documenta en changelog.md  
- **[0,05 pts]** ✅ Hace PR y merge revisado  
- **[0,05 pts]** ✅ Responde issues  

---

# 3. Tester QA/JS - Testing Avanzado (máx. 2,5 puntos)

## A) Testing funcional JavaScript (1,0 puntos)

- **[0,25 pts]** ⚠️ Crea /js/test/api.spec.js con:  
  - Test de respuesta exitosa  
  - Test de error HTTP  
  - Test de error de red  
  - Test de procesamiento con map/filter/reduce  

- **[0,25 pts]** ⚠️ Crea /js/test/library.spec.js con:  
  - Test de inicialización  
  - Test de configuración  
  - Test de funcionalidad principal  
  - Test de manejo de errores  
  > 💬 *Comentario: Ninguno de estos suites fue documentado, ¿corrieron bien? no lo se *

## B) Auditorías Lighthouse (0,9 puntos)

- **[0,15 pts]** ⚠️ Crea baseline test-case-11 (performance ≥80, accessibility ≥90)  
- **[0,15 pts]** ⚠️ Crea post-fetch test-case-12 con comparación  
- **[0,15 pts]** ⚠️ Crea post-library test-case-13 con análisis evolutivo 
 > 💬 *Comentario:  No se reviso la performance en mobile, la cual es baja. no se completaron las observaciones* 

## C) Gestión de issues (0,3 puntos)

- **[0 pts]** ❌ Crea issues por cada problema  
- **[0 pts]** ❌ Asigna issues correctamente  
 > 💬 *Comentario:  Solo se creo una issue https://github.com/Thallys8/turismo-buenos-aires/issues/124, que no se asigno y no se resolvio nunca con ninguna pr asociada. Tu rol es generar issues a partir de todos los test generados, para mejorar la calidad del producto final* 
 
## D) Participación técnica (0,2 puntos)

- **[0,05 pts]** ✅ Crea rama feature/tester-qa-js-testing-suite  
- **[0,05 pts]** ✅ Documenta en changelog.md  
- **[0 pts]** ❌ Actualiza testing-doc.md  
- **[0,05 pts]** ✅ Hace PR y merge revisado  

---

# 4. Coordinador/DevOps – Gestión y Resolución de Correcciones (máx. 2,5 puntos)

## FASE 1 – Durante el desarrollo (1,0 punto)

### Correcciones Actividad N°4 (0,3 puntos)

- **[0,15 pts]** ✅ Revisa Request Changes y coordina correcciones  
- **[0,15 pts]** ✅ Realiza backport hacia develop  

### Coordinación del equipo (0,4 puntos)

- **[0,1 pts]** ✅ Mantiene estructuras y actualiza README  
- **[0,05 pts]** ⚠️ Revisa mínimo 3 PRs  
- **[0,1 pts]** ✅ Asegura actualización de changelog  
- **[0,1 pts]** ✅ Administra issues (Kanban)  
 > 💬 *Comentario:  Hubo pocas observaciones claras para algunas PR, por lo que hubo roles que subieron cosas muy incompletas a develop* 

### Preparación de entrega (0,3 puntos)

- **[0,1 pts]** ✅ Crea rama release/segundo-parcial  
- **[0,05 pts]** ⚠️ Configura GitHub Pages  
- **[0,1 pts]** ✅ Publica PR en Slack y campus  
 > 💬 *Comentario:  No se publico en GPages para la entrega* 
---

## FASE 2 – Resolución de Correcciones (1,5 puntos)

### Gestión de Request Changes (0,6 puntos)

- **[- pts]** ⌛ Analiza todos los Request Changes  
- **[- pts]** ⌛ Crea issues específicas y asigna  
- **[- pts]** ⌛ Coordina aplicación de correcciones  

### Proceso iterativo de correcciones (0,5 puntos)

- **[- pts]** ⌛ Aplica correcciones con commits descriptivos  
- **[- pts]** ⌛ Notifica con ✅ cada comentario resuelto  
- **[- pts]** ⌛ Mantiene comunicación fluida  

### Finalización exitosa (0,4 puntos)

- **[- pts]** ⌛ Obtiene LGTM antes de la fecha límite  
- **[- pts]** ⌛ Merge release → master  
- **[- pts]** ⌛ Backport a develop  
---

# Opciones de emojis para evaluación:
- ✅ Completado correctamente  
- ⚠️ Completado con observaciones  
- ❌ No completado  

---

# Puntaje Total  
**Total: 5 / 10 puntos**

---

# 📊 Tabla de Notas Finales del Segundo Parcial

| **Estudiante** | **Act. Obligatoria N°3** | **Act. Obligatoria N°4** | **Segundo Parcial** | **Performance Individual** | **Nota Final** |
|---|---|---|---|---|---|
| Thallys Gomes Leandro | 2 | 6 | 5 | ⌛ | **⌛ > 💬 *Comentario:  @Thallys8 Tu nota de parcial va quedar pendiente, vas a estar asignado a resolver TODAS las request changes de esta entrega, hacer el merge a master y backport a develop, con la fecha y hora limite previamente establecida en la consigna. De no cumplir esto, vas a tener que rendir en la instancia de recuperatorio. Esto debido a que tu aporte al parcial fue muy bajo y el desempeño en la actividad obligatoria n°3 fue baja* ** |
| Guardo Tomas Agostino  | 4 | 6 | 5 | = | **5(Cinco)** |

---

> **Nota:**  
El Segundo Parcial sienta las bases para el **Trabajo Final**, que incluye las cuatro pre-entregas corregidas. La retroalimentación en esta etapa es clave, ya que en el Final deberán agregar una **funcionalidad adicional** y realizar una **defensa oral**.

Thallys8 committed:
Reenplazando archivo apiService.js por un archivo local atraciones.json

Thallys8 committed:
Actualizando errores ortograficos en atracciones.json y adaptando script.js y models/ConexiinAlmacen.js para usar las informciones de api/atraciones.json

Thallys8 committed:
adecuando a js/models/FiltroAtraciones.js a los cambios de fetch para integracion con js/models/ConexionAlmacen.js y js/api/atraciones.json

Thallys8 committed:
configurando el filtro model en js/models/FiltroAtraciones.js

Thallys8 committed:
Adecuamos el código de js/test/api.spec.js para reflejar las pruebas de js/api/atracciones.json

Thallys8 committed:
corrigiendo códgos de las pruebas jasmine

Thallys8 committed:
completando exitosamente la prueba de los moselos.spec

Thallys8 committed:
pruebas de storage.spec

Thallys8 committed:
Completando pruebas jasmine

Thallys8 committed:
correciones se modelos y script

MVelasquez98 reviewed (changes requested):
@Thallys8 No cumpliste con lo solicitado para el Segundo Parcial en el periodo establecido, el mismo esta desaprobado, vas a tener que presentarte en recuperatorio:

> ⚠️ CRÍTICO - PERÍODO DE CORRECCIONES:
> Plazo: Desde la fecha de corrección del parcial hasta el 28 de noviembrea las 22:00 hs.
> Proceso de corrección iterativo:
> Recepción de Request Changes:
> El docente realizará code review en la PR release/segundo-parcial → master
> Se marcarán "Request Changes" con comentarios específicos sobre qué corregir
> Pueden ser sobre código, documentación, estructura, testing, etc.
> Coordinación de correcciones:
> El Coordinador/DevOps debe analizar cada Request Change
> Determinar qué rol del equipo debe resolver cada corrección
> Crear issues específicas para cada corrección solicitada
> Asignar issues a los integrantes correspondientes
> Establecer prioridades (crítico, alto, medio, bajo)
> Aplicación de correcciones:
> Las correcciones se aplican sobre la misma rama release/segundo-parcial
> Cada corrección debe hacerse con commits descriptivos
> Ejemplo: fix: corregir manejo de errores en fetch según review #45
> Notificación al docente:
> Una vez resueltos todos los comentarios, notificar en el hilo de Slack con ✅ :white_check_mark
> Proceso iterativo:
> El docente revisará nuevamente
> Puede haber múltiples rondas de correcciones
> El proceso se repite hasta obtener el LGTM (Looks Good To Me) y "Approve" final
> Aprobación final:
> Una vez obtenido el LGTM del docente en GitHub
> El Coordinador/DevOps realiza el merge pull request de release/segundo-parcial → master
> Notifica en Slack con :pr-merged: 
> Creación de Release con Tag: Una vez aprobado y realizado el merge a master:
> Ir a [Releases](https://docs.github.com/es/repositories/releasing-projects-on-github/managing-releases-in-a-repository) → Draft a new release en GitHub.
> Crear un [tag](https://docs.github.com/es/repositories/releasing-projects-on-github/viewing-your-repositorys-releases-and-tags) siguiendo el formato: v2.0-segundo-parcial
> En el cuerpo del release copiar el changelog correspondiente (extracto de changelog.md).
> Publicar el release.
> Backport final (CRÍTICO):
> Realizar backport de release/segundo-parcial hacia develop:
> Notificar en el hilo de Slack con :pr-merged: "BACKPORT completado"
> CONSECUENCIAS DEL INCUMPLIMIENTO
> ⚠️ ADVERTENCIA CRÍTICA:
> Si el Coordinador/DevOps NO logra resolver todas las Request Changes antes de la fecha límite establecida:
> ❌ Desaprobará automáticamente el Segundo Parcial
> ❌ Deberá rendir recuperatorio
> ❌ Esto ocurre independientemente de:
> La calidad del trabajo inicial entregado
> La nota obtenida por el resto del equipo
> El esfuerzo realizado durante el desarrollo
> Razón de esta política:
> El rol de Coordinador/DevOps es el responsable de la entrega final del producto
> En un entorno profesional, no entregar correcciones a tiempo tiene consecuencias directas
> Esta responsabilidad simula la presión real de un líder técnico en producción

Tomig02 committed:
Merge pr #128 feature/correction-dev-async-fetch-api

Feature/correction dev async fetch api

Tomig02 committed:
Merge branch 'release/segundo-parcial' into feature/correction-tester-qa-js-testing-suite

Thallys8 committed:
Merge pull request #129 from Thallys8/feature/correction-tester-qa-js-testing-suite

Feature/correction tester qa js testing suite

Thallys8 commented:
Pude identificar el error y corregir el código de Test-runner
tengo todas las pruebas hechas con suceso,
solo me quedó comletar el testing-doc.md y completar las pruebas de mobile de Lighthouse

MVelasquez98 commented on js/script.js:
```diff
@@ -7,12 +6,16 @@ import Semana from './models/Semana.js';
 import Validador from './models/Validador.js';
 
 const conexionAlamacen = new ConexionAlamacen();
+await conexionAlamacen.ready;   // esperamos a que atracciones.json se cargue
```
Esta no es la forma ¿por que borraste el archivo apiService.js? Es el intermediario para llamar mediante fetch el consumo de los JSON. No de esta forma

MVelasquez98 reviewed (changes requested):
Tenes que resolver todas las request changes, junto con el backport y versionado final, como indica la consigna del recuperatorio con fecha limite el 3/12. De lo contrario la materia no esta aprobada

Thallys8 committed:
agregando nuevamente el archivo apiService.js

Thallys8 committed:
corrigiendo código de apiService.js y adaptado a lal cambio de api externa para el archivo atracciones.json

Thallys8 committed:
agregando captura de pantalla de las pruebas jasmine

Thallys8 committed:
agregando informaciones de los test Jasmine, completando las tablas

Thallys8 committed:
marcando en el código en donde paré para seguir más tarde

Thallys8 committed:
completamos el documento testing-doc.md

Thallys8 committed:
Cargando las puebas LightHouse Mobile en el tast case 13

Thallys8 committed:
Merge pull request #131 from Thallys8/feature/correction-tester-qa-js-testing-suite

Com ese es el Recuperiatório, ya no estamos más en equipo, por lo tanto no tengo como solicitar revisión de mi compañero.

MVelasquez98 reviewed (changes requested):
Por favor revisa los comentarios sin resolver y marca resueltos los que crees que ya solucionaste. No estas usando fetch, no estas cargando correctamente el JSON estatico

Thallys8 committed:
Merge branch 'feature/correction-dev-async-fetch-api' of https://github.com/Thallys8/turismo-buenos-aires into feature/correction-dev-async-fetch-api

Thallys8 committed:
Se borró espácios en el tezto de la tabla para que las barras estén alinãnas, cambio esttico del código, no altera la funcionalidad

Thallys8 committed:
adaptando el código script.js para usar la apiService mediante fech

Thallys8 committed:
Update changelog.md

Thallys8 committed:
Merge pull request #132 from Thallys8/feature/correction-dev-async-fetch-api

Como ese es el recuperatório, no tengo un equipo para solicitar la revisión de los compañeros.

Thallys8 committed:
Update changelog with recent changes and fixes

Thallys8 committed:
Fix README, changelog, and index files

Updated README, changelog, and index files to align with the partial delivery.

Thallys8 committed:
Update README.md

Actualizando readme para los obdetivos del recuperatório de parcial 2

Thallys8 committed:
Update changelog.md

Thallys8 committed:
Merge pull request #133 from Thallys8/feature/correction-dev-async-fetch-api

Feature/correction dev async fetch api

Correción de Readme

MVelasquez98 commented on js/api/atracciones.json:
```diff
@@ -0,0 +1,52 @@
+{
```
Tenes errores por consola:
<img width="563" height="188" alt="image" src="https://github.com/user-attachments/assets/b1b36e68-aec9-435b-9f53-ee58739dda81" />

MVelasquez98 commented on js/api/apiService.js:
```diff
@@ -0,0 +1,224 @@
+const API_URL = "./js/api/atracciones.json";
+
+/** Sanitiza un string: asegura que sea string y recorta espacios. */
+function sanitizeString(value, fallback = "") {
```
No porbaste estas funciones nuevas en el archivo api.spec.js con test unitarios por cada función, a su vez necesitas actualizar la doc de testing-doc.md

Thallys8 committed:
adaptando ConexionAlmacen para corregir errores de consola

Thallys8 committed:
nuevo intento de corriir error de consola de conexionAlmacen

Thallys8 committed:
Merge pull request #134 from Thallys8/feature/correction-dev-async-fetch-api

Feature/correction dev async fetch api
Correción de js/models/ConexiónAlmace.js para corrigir errores de consola.

Thallys8 committed:
Corrigiendo las pruebas de test runner api

Thallys8 committed:
corrección de errores del test script.spec.js, correciones relacionadas al Flujo 1

Thallys8 committed:
Completando test script,js, correción de errores del flujo 4

Thallys8 committed:
Completamos en el  testing-doc las correciones correpondientes a los cambios de script.js

Thallys8 committed:
Merge pull request #135 from Thallys8/feature/correction-tester-qa-js-testing-suite

Feature/correction tester qa js testing suite

MVelasquez98 commented on js/script.js:
```diff
@@ -93,36 +95,55 @@ function generarMenuReserva(event){
+    try {
+        // 1) Pedimos los datos al apiService (fetch)
+        // Usamos window.obtenerAtracciones para poder mockearla en los tests
+        const todasLasAtracciones = await window.obtenerAtracciones();
```
¿Por que usas el objeto window? Tenes que usar la funcion exportada de apiService.js

Thallys8 committed:
corrigiendo script para remover window

Thallys8 committed:
Merge branch 'release/segundo-parcial' into feature/correction-dev-async-fetch-api

Thallys8 committed:
Merge pull request #136 from Thallys8/feature/correction-dev-async-fetch-api

corrigiendo script para remover window

Thallys8 committed:
correción de Filtro atraciones para identificar los archivos JSON y conectar con la busqueda

Thallys8 committed:
Merge branch 'feature/correction-dev-async-fetch-api' of https://github.com/Thallys8/turismo-buenos-aires into feature/correction-dev-async-fetch-api

Thallys8 committed:
Corrigiendo Filtroatraciones.js , script.js para que funcione el código, el index.html fue solamente una organización visual del código existente.

Thallys8 committed:
borrando comlpetamente window dw script.js

Thallys8 committed:
-

Thallys8 committed:
corrigiendo tarjetasMTML

Thallys8 committed:
Ingresando atributos de las clases que faltaban

Thallys8 committed:
Merge pull request #137 from Thallys8/feature/correction-dev-async-fetch-api

Feature/correction dev async fetch api

Thallys8 committed:
corrigiendo el camino de las imagenes de imsSrc de las atracciones

Thallys8 committed:
corrigiendo el camino de las imagenes de imsSrc de las atracciones

MVelasquez98 commented on js/script.js:
```diff
@@ -496,91 +515,120 @@ function generarMenuItinerario(opciones) {
+
+    const imagenHTML = imgSrc
+        ? `<img loading="lazy" src="${imgSrc}" alt="${datosAtraccion.altFoto || titulo}" >`
+        : "";
```
Nunca se ve la imagen ni descripciones

MVelasquez98 commented on js/api/atracciones.json:
```diff
@@ -0,0 +1,88 @@
+{
+  "atracciones": [
+    {
+      "nombreAtraccion": "elCaminito",
```
¿Por qué en camelCase los nombres?
Respeta las actividades y nombres como se ve en el mockup:
<img width="1027" height="636" alt="image" src="https://github.com/user-attachments/assets/0db13101-b728-431a-aaad-3f7cfcec9ade" />

MVelasquez98 commented on js/script.js:
```diff
@@ -496,91 +515,120 @@ function generarMenuItinerario(opciones) {
-        opcionesAtraccion.push(`<option value="${atracciones[i]}">${atracciones[i]}</option>`);
-    }
-    opcionesAtraccion.push(`<option value="ninguna" selected> Ninguna </option>`);
+        const atracciones = datosAtracciones.map(a => a.nombreAtraccion);
```
Todos los nombres de las opciones estan en camelCase:
<img width="721" height="420" alt="image" src="https://github.com/user-attachments/assets/82c0a245-8a5b-4338-9db1-ede9cf10ba0f" />

MVelasquez98 reviewed (changes requested):
Por favor borra las ramas que no corresponden:
<img width="360" height="486" alt="image" src="https://github.com/user-attachments/assets/14234a7c-2d11-44d4-82e7-0a9dffef7f5e" />

Resolve todas las request changes, recorda que para aprobar el recuperatorio necesitas seguir todavía todos los pasos siguientes:

> Gestión de Request Changes (0,6 puntos):
> [0,2 pts] Analiza todos los Request Changes del docente de forma exhaustiva
> [0,2 pts] Crea issues específicas para cada corrección y las asigna correctamente
> [0,2 pts] Coordina al equipo para aplicar correcciones sobre release/segundo-parcial
> Proceso iterativo de correcciones (0,5 puntos):
> [0,2 pts] Aplica correcciones con commits descriptivos y estructurados
> [0,2 pts] Notifica al docente con ✅ cada vez que se resuelven comentarios
> [0,1 pts] Mantiene comunicación fluida durante todo el proceso
> Finalización exitosa (0,4 puntos):
> [0,2 pts] CRÍTICO: Obtiene LGTM y approve del docente en GitHub antes de la fecha límite
> [0,1 pts] Realiza merge de release/segundo-parcial → master y notifica con :pr-merged:
> [0,1 pts] CRÍTICO: Realiza backport de release/segundo-parcial → develop y notifica.
> Tag y release v2.0-segundo-parcial creados

Y posteriormente subir el enlace de la release y backport al campus con fecha limite  jueves, 4 de diciembre de 2025, 23:55, de lo contrario el recuperatorio esta desaprobado como indica la consigna

Thallys8 committed:
Actualizando changelog

Thallys8 committed:
nuevointento de corregir las targetas graficas

Thallys8 committed:
-

Thallys8 committed:
-

Thallys8 committed:
-

Thallys8 committed:
nuevo intento de corregir las tarjetas de imagenes

Thallys8 committed:
nuevo intento de corregir las tarjetas de imagenes

Thallys8 committed:
nuevo intento de corregir las tarjetas de imagenes

Thallys8 committed:
nuevo intento de corregir las tarjetas de imagenes

Thallys8 committed:
nuevo intento de corregir las tarjetas de imagenes

Thallys8 committed:
volviendo a la ultima vesión que funcionó medianamente

Thallys8 committed:
nuevo intento de corregir las tarjetas de imagenes

Thallys8 committed:
-

Thallys8 committed:
-

Thallys8 committed:
-

Thallys8 committed:
Corrigiendo error en crear reserva

Thallys8 committed:
-

Thallys8 committed:
corrigiendo errores en las reservas

Thallys8 committed:
corrigiendo targetas atracciones

Thallys8 committed:
-

Thallys8 committed:
-

Thallys8 committed:
Update changelog.md

Thallys8 committed:
Merge pull request #138 from Thallys8/feature/correction-dev-async-fetch-api

Feature/correction dev async fetch api

MVelasquez98 commented on js/models/Reserva.js:
```diff
@@ -1,61 +1,78 @@
+
+export default class Reserva {
+    constructor() {
+        /**
```
Pero estas definiendo un objeto vacio como "Reserva", justamente la estructura interna la tenes que representar con [atributos de la clase,](https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Classes/Public_class_fields) no solo con un comentario

MVelasquez98 commented on js/models/Reserva.js:
```diff
@@ -1,61 +1,78 @@
+        // Lógica de precio (inventada, podés ajustarla a tu criterio)
+        // --- EJEMPLO ---
+        // Base por persona
+        let precioBasePorPersona = 5000;
```
¿El precio base no varia segun atraccion? Esto deberia ser un campo del JSON atracciones.json

MVelasquez98 reviewed (changes requested):
Por favor, marca "Resolve conversation" lo que ya resolviste

Thallys8 committed:
Cortamos texto del lotem isum de descrición

Thallys8 committed:
-

Thallys8 committed:
-

Thallys8 committed:
Revise changelog for recent feature updates

Updated changelog with recent pull requests and corrections related to the async fetch API and reservation constructor.

Thallys8 committed:
Merge branch 'release/segundo-parcial' into feature/correction-dev-async-fetch-api

Thallys8 committed:
Merge pull request #139 from Thallys8/feature/correction-dev-async-fetch-api

Feature/correction dev async fetch api

MVelasquez98 commented on js/script.js:
```diff
@@ -496,91 +516,120 @@ function generarMenuItinerario(opciones) {
+    elementoHTML.setAttribute("data-aos", fadeStyle);
+    elementoHTML.setAttribute("data-aos-delay", "400");
+
+    // Usamos campos reales del JSON + defaults
```
Esto sigue sin estar resuelto:

Como se ve ahora en desktop:
<img width="1438" height="789" alt="image" src="https://github.com/user-attachments/assets/57daf820-c4d1-4299-b448-f069d04c0d63" />
Como se ve en el mockup:
<img width="1027" height="636" alt="image" src="https://github.com/user-attachments/assets/0db13101-b728-431a-aaad-3f7cfcec9ade" />
¿Por qué hay un downgrade del diseño? En la entrega anterior estas tarjetas se veian de forma correcta con el diseño propuesto. Tiene que verse como el diseño del mockup, tal como estaban en la entrega anterior, solo que ahora renderizandose a partir del JSON.

- ¿Probaste el uso de [imagenes](https://getbootstrap.com/docs/5.3/content/images/) responsive de Bootstrap?
- ¿Estas usando bien el [sistema de columnas wrapping](https://getbootstrap.com/docs/5.3/layout/columns/#column-wrapping) para el responsive y que el texto no te empuje toda la tarjeta?

MVelasquez98 reviewed (changes requested):
No realizaste ningun cambio en el diseño en las tarjetas, se siguen viendo muy mal.
Esto sigue sin estar resuelto:

Como se ve ahora en desktop:
<img width="1438" height="789" alt="image" src="https://github.com/user-attachments/assets/57daf820-c4d1-4299-b448-f069d04c0d63" />
Como se ve en el mockup:
<img width="1027" height="636" alt="image" src="https://github.com/user-attachments/assets/0db13101-b728-431a-aaad-3f7cfcec9ade" />
¿Por qué hay un downgrade del diseño? En la entrega anterior estas tarjetas se veian de forma correcta con el diseño propuesto. Tiene que verse como el diseño del mockup, tal como estaban en la entrega anterior, solo que ahora renderizandose a partir del JSON.

- ¿Probaste el uso de [imagenes](https://getbootstrap.com/docs/5.3/content/images/) responsive de Bootstrap?
- ¿Estas usando bien el [sistema de columnas wrapping](https://getbootstrap.com/docs/5.3/layout/columns/#column-wrapping) para el responsive y que el texto no te empuje toda la tarjeta?

Thallys8 committed:
-

Thallys8 committed:
-

Thallys8 committed:
-

Thallys8 committed:
-

Thallys8 committed:
-

Thallys8 committed:
Update changelog.md

Thallys8 committed:
Remove duplicate changelog entry for image correction

Removed duplicate entry for image display correction in attraction cards.

Thallys8 committed:
Merge branch 'release/segundo-parcial' into feature/correction-dev-async-fetch-api

Thallys8 committed:
Merge pull request #140 from Thallys8/feature/correction-dev-async-fetch-api

Feature/correction dev async fetch api

MVelasquez98 reviewed (changes requested):
Te esta cortando la tarjeta, no se ve el mapa completo, ni el boton de reservar. ¿revisas en local esto antes de probar en githubpages? 
<img width="1393" height="552" alt="image" src="https://github.com/user-attachments/assets/1b14f43f-78c8-406a-bd7a-5f41e24c7575" />

Thallys8 commented:
Esa no es la taejeta de mi ultima actualizción

Thallys8 committed:
-

Thallys8 committed:
Merge pull request #141 from Thallys8/feature/correction-dev-async-fetch-api

PULL REQUEST – Recuperatório del Parci

Thallys8 committed:
-

Thallys8 committed:
Merge pull request #142 from Thallys8/feature/correction-dev-async-fetch-api

as

Thallys8 committed:
Update components.css

No se actualizó correctamente en las ultimas 2 PRs

MVelasquez98 reviewed (changes requested):
Sigue sin verse bien, no es cuestión de reducir el texto. Es cuestión de entender y aplicar bien responsive:
<img width="1421" height="502" alt="image" src="https://github.com/user-attachments/assets/feaa7159-b071-4809-aeba-bc6134ea8a81" />

Thallys8 committed:
Reduce footer image width from 30% to 20%

Thallys8 committed:
Change tarjeta width from 45% to 55% for larger screens

Thallys8 commented:
Estaba tratando de ajustar con components, pero el problema era en pantallas superiores a 1300px de responsive, ahora lo encontré

Thallys8 committed:
Update responsive.css

MVelasquez98 reviewed (changes requested):
Tenes errores por consola en la reserva:
<img width="488" height="160" alt="image" src="https://github.com/user-attachments/assets/c3fb8426-d108-4bbf-8eff-f71fb142c59b" />

MVelasquez98 reviewed (changes requested):
No se cumplieron todos los puntos solicitados para el recuperatorio en el tiempo establecido. El mismo esta desaprobado
> B) FASE DE CORRECCIÓN (Post-entrega del Segundo Parcial)
> ⚠️ CRÍTICO - PERÍODO DE CORRECCIONES:
> Plazo: Desde la fecha de corrección del parcial hasta el 04 de diciembre a las 23:55 hs.
> Proceso de corrección iterativo:
> Recepción de Request Changes:
> El docente realizará code review en la PR release/segundo-parcial → master
> Se marcarán "Request Changes" con comentarios específicos sobre qué corregir
> Pueden ser sobre código, documentación, estructura, testing, etc.
> Coordinación de correcciones:
> El Coordinador/DevOps debe analizar cada Request Change
> Determinar qué rol del equipo debe resolver cada corrección
> Crear issues específicas para cada corrección solicitada
> Asignar issues a los integrantes correspondientes
> Establecer prioridades (crítico, alto, medio, bajo)
> Aplicación de correcciones:
> Las correcciones se aplican sobre la misma rama release/segundo-parcial
> Cada corrección debe hacerse con commits descriptivos
> Ejemplo: fix: corregir manejo de errores en fetch según review #45
> Notificación al docente:
> Una vez resueltos todos los comentarios, notificar en el hilo de Slack con ✅ :white_check_mark
> Proceso iterativo:
> El docente revisará nuevamente
> Puede haber múltiples rondas de correcciones
> El proceso se repite hasta obtener el LGTM (Looks Good To Me) y "Approve" final
> Aprobación final:
> Una vez obtenido el LGTM del docente en GitHub
> El Coordinador/DevOps realiza el merge pull request de release/segundo-parcial → master
> Notifica en Slack con :pr-merged: 
> Creación de Release con Tag: Una vez aprobado y realizado el merge a master:
> Ir a [Releases](https://docs.github.com/es/repositories/releasing-projects-on-github/managing-releases-in-a-repository) → Draft a new release en GitHub.
> Crear un [tag](https://docs.github.com/es/repositories/releasing-projects-on-github/viewing-your-repositorys-releases-and-tags) siguiendo el formato: v2.0-segundo-parcial
> En el cuerpo del release copiar el changelog correspondiente (extracto de changelog.md).
> Publicar el release.
> Backport final (CRÍTICO):
> Realizar backport de release/segundo-parcial hacia develop:
> Notificar en el hilo de Slack con :pr-merged: "BACKPORT completado"
> Tag y release v2.0-segundo-parcial creados
> CONSECUENCIAS DEL INCUMPLIMIENTO
> ⚠️ ADVERTENCIA CRÍTICA:
> Si el Coordinador/DevOps NO logra resolver todas las Request Changes antes de la fecha límite establecida:
> ❌ Desaprobará automáticamente el Recuperatorio Segundo Parcial
> ❌ Esto ocurre independientemente de:
> La calidad del trabajo inicial entregado
> La nota obtenida por el resto del equipo
> El esfuerzo realizado durante el desarrollo
> Razón de esta política:
> El rol de Coordinador/DevOps es el responsable de la entrega final del producto
> En un entorno profesional, no entregar correcciones a tiempo tiene consecuencias directas
> Esta responsabilidad simula la presión real de un líder técnico en producción

Tomig02 committed:
Agrego valor faltante (precio) en el retorno de la API

Tomig02 committed:
Correccion de error por null en reserva

Tomig02 committed:
texto mas realista de tarjetas y correcciones de responsiveness

Tomig02 committed:
carga lazy del css de AOS y ciertos imports en script + conexionAlmacen

Tomig02 committed:
prefetch de los modulos que son necesarios al inicio

Tomig02 committed:
modificacion del tamaño de imagenes y carga de bootstrap

MVelasquez98 reviewed (changes requested):
@Tomig02 Tenes varios errores por consola por resolver

Tomig02 committed:
Correccion de error en imports por .default faltante

MVelasquez98 reviewed (approved):
@Tomig02 LGTM proseguí con las correcciones de la mesa de examen
